### PR TITLE
fix: [IABT-1438] Implements zendesk Custom fields for payment log

### PR DESCRIPTION
**Pull request by CrisTofani**
## Short description
This PR implements new zendesk custom fields to make the payment log easier to understand on assistance

## List of changes proposed in this pull request
- map new fields ID
- send payment informations on specific fields on any assistance request functions

## How to test
Open a new assistance request from a failed payment source, check the specification to be respected

### CODE REVIEW

File: ts/screens/wallet/PaymentHistoryDetailsScreen.tsx [hunk 1]
```diff
@@ -118,6 +122,26 @@ class PaymentHistoryDetailsScreen extends React.Component<Props> {
...
+    // Add organization fiscal code custom field
+    addTicketCustomField(
+      zendeskPaymentOrgFiscalCode,
+      this.props.route.params.payment.data.organizationFiscalCode
+    );
+    // Add failure custom field
+    addTicketCustomField(
+      zendeskPaymentFailure,
+      this.props.route.params.payment.failure as string
+    );
```
Comment (francescopersico) [inline ts/screens/wallet/PaymentHistoryDetailsScreen.tsx:134]:
```suggestion
    // Add failure custom field if present
    if(this.props.route.params.payment.failure) {
      addTicketCustomField(
        zendeskPaymentFailure,
        this.props.route.params.payment.failure
      );
    }
```

Failure is optional. What you think about setting the custom field only when needed. It also avoid type casting.

File: ts/screens/wallet/payment/NewTransactionSummaryScreen.tsx [hunk 2]
```diff
@@ -478,6 +483,19 @@ const mapDispatchToProps = (dispatch: Dispatch, props: OwnProps) => {
...
       RptIdFromString.encode(rptId)
     );
+    // Add organization fiscal code custom field
+    addTicketCustomField(
+      zendeskPaymentOrgFiscalCode,
+      rptId.organizationFiscalCode
+    );
+    if (O.isSome(error)) {
+      // Add failure custom field
+      addTicketCustomField(zendeskPaymentFailure, error.value as string);
```
Comment (francescopersico) [inline ts/screens/wallet/payment/NewTransactionSummaryScreen.tsx:493]: Same as above.

File: ts/screens/wallet/payment/TransactionErrorScreen.tsx [hunk 3]
```diff
@@ -114,7 +118,18 @@ const requestZendeskAssistanceForPaymentFailure = (
...
+  // Add organization fiscal code custom field
+  addTicketCustomField(
+    zendeskPaymentOrgFiscalCode,
+    rptId.organizationFiscalCode
+  );
+  // Add rptId custom field
+  addTicketCustomField(zendeskPaymentNav, getCodiceAvviso(rptId));
   if (payment) {
+    // Add failure custom field
+    addTicketCustomField(zendeskPaymentFailure, payment.failure as string);
```
Comment (francescopersico) [inline ts/screens/wallet/payment/TransactionErrorScreen.tsx:130]: As above.

**Review by francescopersico — CHANGES_REQUESTED** [review]

File: ts/screens/wallet/payment/TransactionErrorScreen.tsx (hunk 3 above)
Comment (CrisTofani) [reply ts/screens/wallet/payment/TransactionErrorScreen.tsx]: solved in [a7ea040](https://github.com/pagopa/io-app/pull/4431/commits/a7ea0408ac4eda03089d9c1d11fb8a9ad0e9b595)

File: ts/screens/wallet/payment/NewTransactionSummaryScreen.tsx (hunk 2 above)
Comment (CrisTofani) [reply ts/screens/wallet/payment/NewTransactionSummaryScreen.tsx]: solved in [a7ea040](https://github.com/pagopa/io-app/pull/4431/commits/a7ea0408ac4eda03089d9c1d11fb8a9ad0e9b595)

File: ts/screens/wallet/PaymentHistoryDetailsScreen.tsx (hunk 1 above)
Comment (CrisTofani) [reply ts/screens/wallet/PaymentHistoryDetailsScreen.tsx]: solved in [a7ea040](https://github.com/pagopa/io-app/pull/4431/commits/a7ea0408ac4eda03089d9c1d11fb8a9ad0e9b595)

**Review by francescopersico — APPROVED** [review]

**Pull request merged**